### ISSUE TITLE
Parse error with object property

### ISSUE THREAD

**Issue by azw413** [rusty-ecma/RESSA#55]
Hi, I found a parse error :-

thread 'main' panicked at 'Failed to parse JS: UnexpectedToken(Position { line: 38973, column: 27 }, "Found unexpected token: +=; Not at assignment target")', src/main.rs:26:29

This is the code:

return {
                style: {
                  top: r += "px",
                  left: a += "px"
                }
              };

**Comment by FreeMasen** [comment]
Thanks for the bug report. I will try and take a look this weekend. I am curious if you have validate this happens in a smaller file, something like.

```js
let x = function(r) {
  return {
    top: r += "px",
  }
}
```

I only as because line 38,000+ seems like the parser state may have ended up in a bad place because of some previous syntax. I have seen this especially with regular expressions

**Comment by azw413** [comment]
Funny you should mention regex's, I also raised an issue against your regex library which also produced an error so it's entirely possible that one led to the other. I'll try to make some smaller test cases.

**Comment by FreeMasen** [comment]
Thanks for the extra leg work on that! I anticipate that ressa should probably move to using the [non-context aware scanner](https://github.com/rusty-ecma/RESS/blob/master/src/manual_scanner.rs) from ress which is typically what most js AST producing projects do. 

If you do want to keep digging on where the regex is failing, I would encourage you to try and run your regex though ress [which has a few regex edge case tests](https://github.com/rusty-ecma/RESS/blob/master/tests/snippets/main.rs#L99) already. 

Would you be able to share the original file that this is causing problems with?

**Comment by FreeMasen** [comment]
I have confirmed that the regex from your issue on res-regex does parse correctly in `ress` as a regular expression. Unfortunately that means I'll need more context to determine where this error is coming from. 

Any additional context you could provide would be very helpful

**Comment by azw413** [comment]
Hi Robert, I’ve shared the code privately via your gmail address in your profile. Thanks

**Comment by FreeMasen** [comment]
I've been able to pair things down to the the expression that is failing.

```js
var a = function (b) {
  return {
    c: {
      d: b += 1,
    }
  };
}();
```

It seems like the error is happening when in a call expression, so I'll have to figure out why that isn't setting the assignment target correctly

**Comment by FreeMasen** [comment]
After spending most of today debugging this issue, I am still not quite there. The above isn't quite correct, it seems that an object property that contains an assignment gets things in a bad state, so the second object property is where the error is happening. So the actual minimum error script is

```js
var a = {
  b: c += 1,
  d: e += 1,
};
```
However I am having a hard time isolating where the invalid state is getting set. I will continue to try and figure this out over the next few days but will be limited with my working hours. At the very least I am hopeful that I can have a solution by the end of next weekend.

**Comment by FreeMasen** [comment]
At this point I have realized that the failing tests are new since my last commits in this project. For the time being I am going to just disable the test262 tests until I am able to address the non-feature gated breaking changes provided by that test suite.

**Comment by FreeMasen** [comment]
fixed by #56
